batavia/types/FrozenSet.js

```diff
@@ -5,11 +5,266 @@
+    };
+
+    FrozenSet.prototype.__div__ = function(other) {
+        throw new batavia.builtins.NotImplementedError("FrozenSet.__div__ has not been implemented");
+    };
+
+    FrozenSet.prototype.__floordiv__ = function(other) {
+        throw new batavia.builtins.TypeError("unsupported operand type(s) for //: 'frozenset' and '" + batavia.type_name(other) + "'");
+    };
+
+    FrozenSet.prototype.__truediv__ = function(other) {
+        throw new batavia.builtins.TypeError("unsupported operand type(s) for /: 'frozenset' and '" + batavia.type_name(other) + "'");
+    };
+
+    FrozenSet.prototype.__mul__ = function(other) {
+        throw new batavia.builtins.TypeError("unsupported operand type(s) for +: 'frozenset' and '" + batavia.type_name(other) + "'");
+    };
+
+    FrozenSet.prototype.__mod__ = function(other) {
+        throw new batavia.builtins.NotImplementedError("FrozenSet.__mod__ has not been implemented");
+    };
+
+    FrozenSet.prototype.__add__ = function(other) {
+    throw new batavia.builtins.TypeError("unsupported operand type(s) for +: 'frozenset' and '" + batavia.type_name(other) + "'");
+    };
+
+    FrozenSet.prototype.__sub__ = function(other) {
+        throw new batavia.builtins.NotImplementedError("FrozenSet.__sub__ has not been implemented");
+    };
+
+    FrozenSet.prototype.__getitem__ = function(other) {
+        throw new batavia.builtins.NotImplementedError("FrozenSet.__getitem__ has not been implemented");
+    };
+
+    FrozenSet.prototype.__lshift__ = function(other) {
+        throw new batavia.builtins.NotImplementedError("FrozenSet.__lshift__ has not been implemented");
+    };
+
+    FrozenSet.prototype.__rshift__ = function(other) {
+        throw new batavia.builtins.NotImplementedError("FrozenSet.__rshift__ has not been implemented");
+    };
+
+    FrozenSet.prototype.__and__ = function(other) {
+        if (batavia.isinstance(other, [batavia.types.FrozenSet, batavia.types.Set])){
+            var both = [];
+            var iterobj = batavia.builtins.iter([this], null);
+            batavia.iter_for_each(iterobj, function(val) {
+                if (other.__contains__(val)) {
+                    both.push(val);
+                }
+            });
+            return new FrozenSet(both);
+        }
+        throw new batavia.builtins.TypeError("unsupported operand type(s) for &: 'frozenset' and '" + batavia.type_name(other) + "'");
+    };
+
+    FrozenSet.prototype.__xor__ = function(other) {
+        throw new batavia.builtins.NotImplementedError("FrozenSet.__xor__ has not been implemented");
+    };
+
+    FrozenSet.prototype.__or__ = function(other) {
+        throw new batavia.builtins.NotImplementedError("FrozenSet.__or__ has not been implemented");
+    };
+
+    /**************************************************
+     * Set Iterator
+     **************************************************/
+
+    FrozenSet.prototype.SetIterator = function (data) {
```

Comment:
Is it necessary to duplicate this implementation here? Could we borrow the one from Set (since it should be identical, AFAICT)?